analyzers/src/SonarAnalyzer.CSharp/Rules/DoNotDecreaseMemberVisibility.cs

```diff
@@ -27,30 +27,29 @@
 using Microsoft.CodeAnalysis.Diagnostics;
 using SonarAnalyzer.Common;
 using SonarAnalyzer.Helpers;
+using StyleCop.Analyzers.Lightup;
 
 namespace SonarAnalyzer.Rules.CSharp
 {
     [DiagnosticAnalyzer(LanguageNames.CSharp)]
     [Rule(DiagnosticId)]
     public sealed class DoNotDecreaseMemberVisibility : SonarDiagnosticAnalyzer
     {
-        internal const string DiagnosticId = "S4015";
+        private const string DiagnosticId = "S4015";
         private const string MessageFormat = "This member hides '{0}'. Make it non-private or seal the class.";
 
-        private static readonly DiagnosticDescriptor rule =
-            DiagnosticDescriptorBuilder.GetDescriptor(DiagnosticId, MessageFormat, RspecStrings.ResourceManager);
-        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(rule);
+        private static readonly DiagnosticDescriptor Rule = DiagnosticDescriptorBuilder.GetDescriptor(DiagnosticId, MessageFormat, RspecStrings.ResourceManager);
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(Rule);
 
-        protected override void Initialize(SonarAnalysisContext context)
-        {
+        protected override void Initialize(SonarAnalysisContext context) =>
             context.RegisterSyntaxNodeActionInNonGenerated(c =>
                 {
-                    var classDeclaration = (ClassDeclarationSyntax)c.Node;
+                    var classDeclaration = (TypeDeclarationSyntax)c.Node;
                     var classSymbol = c.SemanticModel.GetDeclaredSymbol(classDeclaration);
```

Comment:
Let`s give it a try, but my understanding is that the containingSymbol is actually the DeclaredSymbol for Class and Record declaration syntax nodes. So maybe we can work with that, without retrieving the declared symbol. (Only in case the kind is NamedType.)

Reply:
That's very nice! This was already here so I'm pretty sure that we can make this improvement in other places too.